src/libs/masks.c

```diff
@@ -1712,7 +1711,21 @@ static void _lib_masks_recreate_list(dt_lib_module_t *self)
 
   --darktable.gui->reset;
 
+  dt_gui_widget_reallocate_now(lm->treeview);
+}
+
+static void _lib_masks_recreate_list(dt_lib_module_t *self)
+{
+  dt_lib_masks_t *lm = self->data;
+  dt_lib_gui_queue_update(self);
```

Comment:
The (possibly postponed or ignored) gui update triggered by `dt_lib_gui_queue_update` is for the list of shapes. If the module or panel is hidden, this does not need to be updated.

The `_update_all_properties` call is for the sliders in the collapsible section. These _always_ need to be updated, because they can be manipulated/read using shortcuts/actions.

Reply:
Ok, thanks for clarifying, I miss-read the patch :) So all good to me. I have tested it a bit and it works on my side.